hedera-node/src/main/java/com/hedera/services/state/submerkle/NftAdjustments.java

```diff
@@ -0,0 +1,132 @@
+package com.hedera.services.state.submerkle;
+
+import com.google.common.base.MoreObjects;
+import com.hedera.services.utils.EntityIdUtils;
+import com.hederahashgraph.api.proto.java.NftTransfer;
+import com.hederahashgraph.api.proto.java.TokenTransferList;
+import com.swirlds.common.io.SelfSerializable;
+import com.swirlds.common.io.SerializableDataInputStream;
+import com.swirlds.common.io.SerializableDataOutputStream;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.IntStream;
+
+import static com.hedera.services.utils.MiscUtils.readableNftTransferList;
+import static java.util.stream.Collectors.toList;
+
+public class NftAdjustments implements SelfSerializable {
+
+	private static final Logger log = LogManager.getLogger(NftAdjustments.class);
+
+	private static final int MERKLE_VERSION = 1;
+
+	private static final long RUNTIME_CONSTRUCTABLE_ID = 0xd7a02bf45e103466L;
+
+	private static final long[] NO_ADJUSTMENTS = new long[0];
+
+	public static final int MAX_NUM_ADJUSTMENTS = 1024;
+
+	private long[] serialNums = NO_ADJUSTMENTS;
+
+	private List<EntityId> senderAccIds = Collections.emptyList();
+
+	private List<EntityId> receiverAccIds = Collections.emptyList();
+
+	public NftAdjustments() {
+
+	}
+
+	@Override
+	public long getClassId() {
+		return RUNTIME_CONSTRUCTABLE_ID;
+	}
+
+	@Override
+	public int getVersion() {
+		return MERKLE_VERSION;
+	}
+
+	@Override
+	public void deserialize(SerializableDataInputStream in, int version) throws IOException {
+		serialNums = in.readLongArray(MAX_NUM_ADJUSTMENTS);
+		senderAccIds = in.readSerializableList(MAX_NUM_ADJUSTMENTS, true, EntityId::new);
+		receiverAccIds = in.readSerializableList(MAX_NUM_ADJUSTMENTS, true, EntityId::new);
+	}
+
+	@Override
+	public void serialize(SerializableDataOutputStream out) throws IOException {
+		out.writeLongArray(serialNums);
+		out.writeSerializableList(senderAccIds, true, true);
+		out.writeSerializableList(receiverAccIds, true, true);
+	}
+
+	/* ---- Object --- */
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) {
+			return true;
+		}
+		if (o == null || getClass() != o.getClass()) {
+			return false;
+		}
+		NftAdjustments that = (NftAdjustments) o;
+		return Arrays.equals(serialNums, that.serialNums)
+				&& senderAccIds.equals(that.senderAccIds) && receiverAccIds.equals(that.receiverAccIds);
+	}
+
+	@Override
+	public int hashCode() {
+		int result = Long.hashCode(RUNTIME_CONSTRUCTABLE_ID);
```

Comment:
Not very important, but `hashCode()` is recommended to use only the fields that affect `equals()`.